### ISSUE TITLE
What does the predicted depth signify?

### ISSUE THREAD

**Issue by zendevil** [intel-isl/MiDaS#42]
A "prediction" gives the following:

[[2496.0127 2495.973 2495.9888 ... 855.7698 855.57666 856.0468 ]
[2495.9575 2495.9158 2495.9329 ... 855.4036 855.20917 855.68256]
[2495.9797 2495.9387 2495.9556 ... 855.55426 855.36035 855.83234]
...
[3245.7551 3245.7756 3245.7664 ... 2852.5774 2852.4922 2852.702 ]
[3245.7275 3245.7478 3245.739 ... 2852.4827 2852.397 2852.6072 ]
[3245.7974 3245.8179 3245.809 ... 2852.7156 2852.6309 2852.8398 ]]

What are the units of these numbers m, mm, ft? Of course those numbers aren't disparities (since the images aren't that wide). So what do these numbers represent? How to convert this prediction to actual depth given camera intrinsics?
Thanks

**Comment by ranftlr** [comment]
The prediction is relative inverse depth. For each prediction, there exist some scalars a,b such that a*prediction+b is the absolute inverse depth. The factors a,b cannot be determined without additional measurements.

**Comment by zendevil** [comment]
What additional measurements can determine a and b?

**Comment by ranftlr** [comment]
You'd need to know the absolute depth of at least two pixels in the image to derive the two unknowns. Based on these measurements you could align the predictions to these measurements as done in our SSIMSE loss. A practical solution would require more points to minimize the likelihood of running into degenerate configurations.

**Comment by 3dsf** [comment]
@ranftlr 
I have depth cameras ([l515](https://www.intelrealsense.com/lidar-camera-l515/), [d435i](https://www.intelrealsense.com/depth-camera-d435i/)).  I'm more of a depth enthusiast than technically strong (coding/ML).  Are you saying I could use combine MiDaS and the depth cameras?  What do you mean by SSIMSE loss?  
_Let me know if I can contribute (using the cameras) to this project or a future depth related project._

**Comment by ranftlr** [comment]
Yes, it is possible to combine MiDaS with depth cameras. You'd have to ask the questions what you'd want to gain by doing so, however. Depth cameras, by definition, can more directly sense depth by employing stereo cameras or LiDaRs. MiDaS is a single-image method, e.g. it is most useful for images that where not captured with a specially designed depth camera. As single-image depth is arguably a harder problem the accuracy that you are getting from a depth camera will usually be higher than what is provided by MiDaS.

The SSIMSE loss is defined equations 1 to 4 in the paper. Specifically, equation 4 would give you the scale and shift that optimally aligns the predictions to the measurements in a least-squares sense.

**Comment by ranftlr** [comment]
Closing due to inactivity.

**Comment by nicolasugrinovic** [comment]
Hi, thank you for the release of this great work.

**TLDR**: Does your model account for the non-linear relationship between disparity and depth `depth = a / disparity `?

I want to use your model to get the depth of a know planar object from a monocular image, lets say the ground depth in the image. As the model output is disparity and I need the depth I use the relationship of: `depth = a / disparity `, where a is any constant that corrects the scale. However, as I can see, when I plot the xyz points of the scene (xy pixel locations + z for depth) I get a non-linear depth. As can be seen in the next figure. 

![Selection_068](https://user-images.githubusercontent.com/25443240/117174931-8f102400-adce-11eb-8a6a-f37d80550d37.png)

However, if I directly use the disparity, I have an almost linear relationship between the xyz points (next figure), which is what I expect as the image of the points contain a flat ground. 

![Selection_069](https://user-images.githubusercontent.com/25443240/117175147-ce3e7500-adce-11eb-84b7-7a3b8bb96377.png)

**Comment by nattec** [comment]
Hi, if I have the distance between two points how can I get the absolute inverse depth?

**Comment by puyiwen** [comment]
> Hi, if I have the distance between two points how can I get the absolute inverse depth?

I have the same question, are you solve it?

**Comment by Sangipc** [comment]
I have a few doubts, if you could kindly clear them

1. Could you clarify the number of layers in the depth map produced by the MiDaS model and how these different layers relate to the depth predictions?

2. what is the unit of the depth maps which are produced as output ? (pixel, mm, cm,...)?

3. How can we transform the output of the MiDaS model, which contains relative depth values, back to the original input resolution? This would help establish a pixel-to-depth map relation between the input image and the corresponding depth map.
